runtime/bcverify/vrfyhelp.c

```diff
@@ -256,8 +256,16 @@ buildStackFromMethodSignature( J9BytecodeVerificationData *verifyData, UDATA **s
 		/* In the <init> method of Object the type of this is Object.  In other <init> methods, the type of this is uninitializedThis */
 		if ((J9UTF8_DATA(utf8string)[0] == '<')	&& (J9UTF8_DATA(utf8string)[1] == 'i') && (classIndex != BCV_JAVA_LANG_OBJECT_INDEX)) {
 			/* This is <init>, not java/lang/Object */
-			PUSH(BCV_SPECIAL_INIT | (classIndex << BCV_CLASS_INDEX_SHIFT));
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+			if (J9_ARE_ALL_BITS_SET(romClass->modifiers, J9AccPrimitiveValueType)) { /* This is temporary and will have to be changed when <vnew> is introduced */
```

Comment:
Please use `J9ROMCLASS_IS_PRIMITIVE_VALUE_TYPE(romClass)`, and also at least mention `J9VM_OPT_VALHALLA_NEW_FACTORY_METHOD` in the comment.